Fixing compilation with gcc-4.5

http://bugs.gentoo.org/show_bug.cgi?id=317751

Patch written by Kacper Kowalik <xarthisius.kk@gmail.com>

--- toycars_vehicle_editor/src/VehicleEditorUI.cxx
+++ toycars_vehicle_editor/src/VehicleEditorUI.cxx
@@ -228,7 +228,7 @@
 sprintf(str, "%d", n);
 convexChoice->add(str);
 convexChoice->value(n);
-vehicleObj.getGeometry()->push_back(std::list<Vec2D>::list());
+vehicleObj.getGeometry()->push_back(std::list<Vec2D>());
 spriteView->setConvexChoice(n);
 spriteView->redraw();
 }
--- toycars_vehicle_editor/src/VehicleObject.cpp
+++ toycars_vehicle_editor/src/VehicleObject.cpp
@@ -152,7 +152,7 @@
    short count;
    geometry.clear();
    for (xConvex = xGeometry->FirstChild(); xConvex != NULL; xConvex = xConvex->NextSibling("convex")) {
-      geometry.push_back(std::list<Vec2D>::list());
+      geometry.push_back(std::list<Vec2D>());
       std::list<Vec2D> &convex = geometry.back();
       readPointsFromString(xConvex->FirstChild()->ToText()->Value(), convex, count);
    }
@@ -411,7 +411,7 @@
    short count;
    geometry.clear();
    for (xConvex = xGeometry->FirstChild(); xConvex != NULL; xConvex = xConvex->NextSibling("convex")) {
-      geometry.push_back(std::list<Vec2D>::list());
+      geometry.push_back(std::list<Vec2D>());
       std::list<Vec2D> &convex = geometry.back();
       readPointsFromString(xConvex->FirstChild()->ToText()->Value(), convex, count);
    }
